src/tests/lib/utils/tooltip.utils.spec.ts

```diff
@@ -0,0 +1,170 @@
+  const tooltipHeight = 40;
+
+  const createTargetAndTooltipRects = ({ x, y }) => ({
+    targetRect: {
+      x,
+      y,
+      width: targetWidth,
+      height: targetHeight,
+      top: y,
+      left: x,
+      bottom: y + targetHeight,
+      right: x + targetWidth,
+    },
+    tooltipRect: {
+      x,
+      y,
+      width: tooltipWidth,
+      height: tooltipHeight,
+      top: y,
+      left: x,
+      bottom: y + tooltipHeight,
+      right: x + tooltipWidth,
+    },
+  });
+
+  describe("when the target isn't close to any edge", () => {
+    const { targetRect, tooltipRect } = createTargetAndTooltipRects({
+      x: containerRect.x + 200,
+      y: containerRect.y + 200,
+    });
+    const scrollbarWidth = 0;
+
+    it("should position the tooltip centered below the target", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+        }),
+      ).toEqual({
+        x: (targetRect.width - tooltipRect.width) / 2,
+        y: targetRect.height,
+      });
+    });
+
+    it("should position the tooltip centered above the target when 'top' is true", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+          top: true,
+        }),
+      ).toEqual({
+        x: (targetRect.width - tooltipRect.width) / 2,
+        y: -tooltipRect.height,
+      });
+    });
+  });
+
+  describe("when the target is close to the left edge", () => {
+    const { targetRect, tooltipRect } = createTargetAndTooltipRects({
+      x: containerRect.x + 10,
+      y: containerRect.y + 200,
+    });
+    const scrollbarWidth = 0;
+
+    it("should position the tooltip touching the left edge", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+        }),
+      ).toEqual({
+        x: -10,
+        y: targetRect.height,
+      });
+    });
+
+    it("should position the tooltip centered when 'center' is true", () => {
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+          center: true,
+        }),
+      ).toEqual({
+        x: (targetRect.width - tooltipRect.width) / 2,
+        y: targetRect.height,
+      });
+    });
+  });
+
+  describe("when the target is close to the right edge", () => {
+    const { targetRect, tooltipRect } = createTargetAndTooltipRects({
+      x: containerRect.right - tooltipWidth + 100,
+      y: containerRect.y + 200,
+    });
+
+    it("should position the tooltip touching the right edge", () => {
+      const scrollbarWidth = 0;
+      expect(
+        translateTooltip({
+          containerRect,
+          targetRect,
+          tooltipRect,
+          scrollbarWidth,
+        }),
+      ).toEqual({
+        x: -100,
```

Comment:
Ditto about the 100 in the setup and using a variable instead.

Reply:
I couldn't think of a good name before. But it's done now.